### ISSUE TITLE
Additional exported elements

### ISSUE THREAD

**Issue by whittaker007** [watergis/maplibre-gl-export#21]
Is it possible to add other elements to the export such as attributions, feature keys etc?

**Comment by johnsalzbrunn** [comment]
Also interested in this enhancement - particularly a 'legend' element. Any update on this?

**Comment by zbyte64** [comment]
We do this using html2canvas. It would be nice to see a community owned project provide this functionality.

**Comment by JinIgarashi** [comment]
@zbyte64 Thank you for the comment. Please feel free to create a PR to this repo. I am happy to merge it.

**Comment by zbyte64** [comment]
I'm a bit hesitant to make a PR, mainly concerned over how it changes the API interface being offered. Here are some broad strokes though so we can get a concrete discussion going (and to help others):

We have an async function that calls html2canvas to copy everything from the "map container" (this includes the Canvas as well as the mapbox controls) into a new canvas element:
```ts
async function draw(): Promise<HTMLCanvasElement> {
      return html2canvas(map.getContainer(), {
        ignoreElements,
      });
    }
```

`html2canvas` takes a function that lets us filter out elements we don't want (ie interactive elements). We could let the user pass in a function or make some convenience factory that filters by classnames:
```ts
export function ignoreElementsFactory(hiddenClassNames: string[]) {
  const hiddenClassNamesSet = new Set(hiddenClassNames);
  return (element: Element) => {
    // Remove all elements with class in hiddenClassNames
    // eslint-disable-next-line no-restricted-syntax
    for (const hiddenClassName of element.classList) {
      if (hiddenClassNamesSet.has(hiddenClassName)) {
        return true;
      }
    }
    return false;
  }
}
```

Also, most of the time the attribution isn't optimized for print (ie might be underlined to indicate it's a link), so we may want to also allow the user to easily provide additional HTML elements.

To recap, my questions are:
* how should the user specify which elements to ignore? is there any affordances we want to add (eg filter by class names)?
* how should the user specify elements to inject? maybe a callback to generate html elements, which then get called to html2canvas and blitted over the map result?

**Comment by JinIgarashi** [comment]
@zbyte64, @whittaker007, @johnsalzbrunn , I have managed to add attribution text as a symbol layer into the bottom-right of exported image via https://github.com/watergis/maplibre-gl-export/pull/146
